craft_providers/lxd/lxc.py

```diff
@@ -1197,17 +1197,26 @@ def is_pro_enabled(
         ]
 
         try:
-            proc = self._run_lxc(command, capture_output=True, project=project)
-            data = json.loads(proc.stdout)
-            if data.get("result") == "success":
-                if data.get("data", {}).get("attributes", {}).get("is_attached"):
+            proc = self._run_lxc(
+                command, capture_output=True, check=False, project=project
+            )
+            if proc.returncode == 0:
+                data = json.loads(proc.stdout)
+                if data.get("result") == "success" and data.get("data", {}).get(
+                    "attributes", {}
+                ).get("is_attached"):
                     logger.debug("Managed instance is Pro enabled.")
                     return True
-                logger.debug("Managed instance is not Pro enabled.")
-                return False
+                if data.get("result") == "success":
+                    logger.debug("Managed instance is not Pro enabled.")
+                    return False
```

Comment:
pedantic: this could be a bit more readable by nesting inside `if data.get("result") == "success":`.

```python
if data.get("result") == "success":
    if data.get("data", {}).get("attributes", {}).get("is_attached"):
        logger.debug("Managed instance is Pro enabled.")
        return True
    logger.debug("Managed instance is not Pro enabled.")
    return False
```